storage/invoke_test.go

```diff
@@ -188,7 +218,7 @@ func TestInvoke(t *testing.T) {
 				headers := callctx.HeadersFromContext(ctx)
 				gotClientHeader = headers["x-goog-api-client"][0]
 				gotIdempotencyHeader = headers["x-goog-gcs-idempotency-token"][0]
-				if counter <= test.count {
+				if (counter <= test.count) || (test.retry != nil && counter == test.retry.maxRetryCount) {
```

Comment:
This seems like it shouldn't be necessary? Shouldn't the call to run cut retries off at the correct count without this?